pkg/remote/codec/default_codec.go

```diff
@@ -176,6 +258,11 @@ func (c *defaultCodec) DecodeMeta(ctx context.Context, message remote.Message, i
 		if flagBuf, err = in.Peek(2 * Size32); err != nil {
 			return perrors.NewProtocolErrorWithErrMsg(err, fmt.Sprintf("ttheader read payload first 8 byte failed: %s", err.Error()))
 		}
+		if c.crc32Check && crc32cTable != nil {
```

Comment:
Better check by default as long as the crc32c hash exists in the header.

Reply:
after discussion, no check if not configured.